client.go

```diff
@@ -22,8 +22,25 @@ package cluster
 
 import (
 	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3x/watch"
 )
 
+// A Shard represents a piece of data owned by the service
+type Shard interface {
+	ID() uint32 // ID returns the id of the shard
+}
+
+// NewShard returns a new Shard
+func NewShard(s uint32) Shard { return &shard{id: s} }
```

Comment:
Technically it will still be a heap allocation as the interface type is allocated on the heap and will hold a pointer to the struct.  Still better though as it's not a pointer to a pointer so one less mem access when retrieving. Heh.